crates/typst-cli/src/compile.rs

```diff
@@ -118,6 +118,25 @@ pub fn compile_once(
             print_diagnostics(world, &[], &warnings, command.common.diagnostic_format)
                 .map_err(|err| eco_format!("failed to print diagnostics ({err})"))?;
 
+            if let Some(ref makefile_deps) = command.makefile_deps {
+                // Validation in main.rs ensures --makefile-deps can only be
+                // specified when output is provided and is not stdout.
+                let Some(Output::Path(ref output)) = command.output else {
+                    unreachable!()
+                };
+                let root = world.root().to_path_buf();
+                create_makefile_deps(
+                    makefile_deps,
+                    output,
+                    world.dependencies().map(|dep| {
+                        dep.strip_prefix(&root).map(Path::to_path_buf).unwrap_or(dep)
+                    }),
+                )
```

Comment:
I think the logic should as much as possible be moved into `create_makefile_deps`.

Reply:
Ok, I've done that in the latest commit.